src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/TypeNameObject.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Frozen;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Language;
+
+internal readonly struct TypeNameObject
+{
+    private static readonly ImmutableArray<string> s_knownTypeNames;
+    private static readonly FrozenDictionary<string, int> s_typeNameToIndex;
+
+    private static readonly int s_booleanIndex;
+    private static readonly int s_stringIndex;
+
+    static TypeNameObject()
+    {
+        var knownTypeNames = ImmutableArray.CreateBuilder<string>();
+        var typeNameToIndex = new Dictionary<string, int>(StringComparer.Ordinal);
+
+        Add<object>("object");
+        s_booleanIndex = Add<bool>("bool");
+        Add<int>("int");
```

Comment:
>Add<int>("int"); [](http://example.com/codeflow?start=8&length=16)

nint isn't needed too, right?

Reply:
I used the existing set of simple names Razor looks for. It hasn't been updated in awhile to include things like "nint" or "nuint", but updating it probably be a different change.